### ISSUE TITLE
Alternative blog layout small bug when infinite scroll is used

### ISSUE THREAD

**Issue by rodica-andronache** [Codeinwp/hestia#55]
1. Enable the alternative blog layout
2. Enable infinite scroll for the blog page
-> On the Blog page, when you scroll the page to get the second iterations of posts, the alternative layout is not working properly, causing two posts to have the same position for the featured image.
![Annotation on 2023-03-20 at 12-08-27](https://user-images.githubusercontent.com/3266155/226309587-0e96107f-a0de-48f8-979c-3dc594c48669.png)

**Comment by girishpanchal30** [comment]
Hey @rodica-andronache,

I've reviewed and fixed it with this PR https://github.com/Codeinwp/hestia-pro/pull/251, please check and do let me know if you have any issues.

Thanks

**Comment by rodica-andronache** [comment]
@girishpanchal30 everything's working well 👍

**Comment by HardeepAsrani** [comment]
@girishpanchal30 Can you take a look at why it's not working here? https://github.com/Codeinwp/hestia-pro/pull/2530

**Comment by girishpanchal30** [comment]
@HardeepAsrani I tested with the development branch and it seems working for me, Can you please share your test instance details here?

**Comment by HardeepAsrani** [comment]
Can you share it @rodica-andronache? Thanks both!

**Comment by rodica-andronache** [comment]
@girishpanchal30 here is it: 
`https://tonight-litigant-8ba66c.vertisite.cloud/blog/
user: test
pass: Z3^I7(3XaieXRbWZX#@y5ptT`

Just to be sure, this is the archive I've used https://vertis.d.pr/v/equJsk. Let me know if there's anything I'm missing. Thanks!

**Comment by girishpanchal30** [comment]
Hey @rodica-andronache,

Thank you for sharing the details, I've checked with your test instance and made some changes to fix this issue.
Please check with this PR - https://github.com/Codeinwp/hestia-pro/pull/2533 and do let me know if you have any issues.

Thanks

**Comment by rodica-andronache** [comment]
@girishpanchal30 now, it's working well when there's a sticky post, but now that I deleted that sticky post, the issue is still visible. You can see it there https://tonight-litigant-8ba66c.vertisite.cloud/blog/

**Comment by girishpanchal30** [comment]
@rodica-andronache Fixed, Please check with the latest commit.

**Comment by rodica-andronache** [comment]
@girishpanchal30 it's still not working when there's a sticky post. Plus, there is a post that repeats itself, the last one of the first page, in the case here https://tonight-litigant-8ba66c.vertisite.cloud/blog/ it's `Appearance guide`

**Comment by girishpanchal30** [comment]
@rodica-andronache Fixed, please do some more testing with the latest commit and let me know if you find out any issues.

**Comment by rodica-andronache** [comment]
@girishpanchal30 still not working, with 2 posts per page https://vertis.d.pr/i/0JPKaZ the posts get duplicated  https://vertis.d.pr/i/GIAuUL

**Comment by girishpanchal30** [comment]
@rodica-andronache Fixed the mentioned issue, Can you please check with the latest commit?

**Comment by rodica-andronache** [comment]
@girishpanchal30 it's the same, not working. If you find it more easy, you can use the details that I've shared before, to update the theme there and check yourself
`https://tonight-litigant-8ba66c.vertisite.cloud/blog/ user: test pass: Z3^I7(3XaieXRbWZX#@y5ptT`

**Comment by girishpanchal30** [comment]
@rodica-andronache I can't see any duplicate with latest commit, can you please check with the private window? it might be a cache issue.
Ref: https://www.loom.com/share/d131c05855cc460789900cbb396fbde1

**Comment by rodica-andronache** [comment]
@girishpanchal30 you are right, it was caching probably. I'll check the rest of the things and let you know if I still see anything else. Thanks!

**Comment by rodica-andronache** [comment]
@girishpanchal30 I've renamed my posts like this to know exactly their order https://vertis.d.pr/i/oBP86x . If you go ahead and change the number of posts per page to different values, the order of the posts is different each time https://www.loom.com/share/e5e161d8ae8642f0ab54008e1d1e36a4

**Comment by girishpanchal30** [comment]
@rodica-andronache Fixed, please check with the latest commit, and do let me know if you have any issues.

**Comment by rodica-andronache** [comment]
@girishpanchal30 when we use Featured posts https://vertis.d.pr/i/lUpoqX, some of the posts get duplicated depending on the number of posts per page set up in Settings https://vertis.d.pr/i/jmLS2Y

**Comment by HardeepAsrani** [comment]
Hey @girishpanchal30,

Can you test on all the different usecases before putting it for QA? It might be easier that way instead of the back & forth. If it helps, you can set up a test site with different scenarios and ensure they all work correctly. Please let me know if you need any help.

**Comment by girishpanchal30** [comment]
@HardeepAsrani Yes sure

**Comment by girishpanchal30** [comment]
Hey @rodica-andronache,

Fixed the issue with featured posts and now all the things are working well.
Can you please recheck with the latest commit?

Thanks

**Comment by rodica-andronache** [comment]
@girishpanchal30 https://tonight-litigant-8ba66c.vertisite.cloud/blog/ there are posts missing. It jumps from 1,2,3 ( featured posts) to 8,9 etc https://vertis.d.pr/v/7ZVY7l

**Comment by girishpanchal30** [comment]
Hey @rodica-andronache,

Post missing, duplicate posts and the featured posts-related issue is fixed with the latest commit, please check with the private window and do let me know if you have any issues.

Thanks

**Comment by rodica-andronache** [comment]
hi @girishpanchal30 when using featured posts, with a number of posts per page grater than 3, some posts are missing.
For example, if posts per page is 4 https://vertis.d.pr/i/rgdxBO, the posts listed are 1,2,3,5.. https://vertis.d.pr/i/YRc6F5
If posts per page is 5, the posts listed are 1,2,3, 6..
If posts per page is 6, the posts listed are 1,2,3, 7..

**Comment by HardeepAsrani** [comment]
Hey @rodica-andronache @girishpanchal30, given that this is taken more time than expected, we can put this on hold for now as to avoid a fix leading to more regressions in the future. For now, we will revert changes for this PR in the development branch and tackle it at some other point. Thank you both for your work.
